index.js

```diff
@@ -479,14 +487,14 @@ module.exports = class Hypercore extends EventEmitter {
     if (this.replicator !== null) {
       this.replicator.findingPeers -= this._findingPeers
       this.replicator.clearRequests(this.activeRequests, err)
-      this.replicator.updateActivity(this._active ? -1 : 0, this)
+      this.replicator.updateActivity(this._active ? -1 : 0)
     }
 
     this._findingPeers = 0
 
-    if (this.sessions.length) {
+    if (this.sessions.length || this.core.active > 0) {
       // if this is the last session and we are auto closing, trigger that first to enforce error handling
-      if (this.sessions.length === 1 && this.autoClose) await this.sessions[0].close(err)
+      if (this.sessions.length === 1 && this.core.active === 1 && this.autoClose) await this.sessions[0].close(err)
```

Comment:
It's not clear to me how the case where `this.core.active > 1` and `this.sessions.length === 1` is handled (or if it can occur). In that case, is that session guaranteed to be closed later?

Reply:
Okay, makes sense (assuming you meant `replicator.js` instead of core.js, but 99% sure you did)